app/models/user/academics/registrations.rb

```diff
@@ -14,11 +14,23 @@ def all
       end
 
       def term_ids
-        all.map { |attr| attr.term_id }.compact.uniq
+        all.map { |reg| reg.term_id }.compact.uniq
       end
 
       def find_by_term_id(term_id)
-        all.select { |attr| attr.term_id == term_id }
+        all.select { |reg| reg.term_id == term_id }
+      end
+
+      def latest_academic_level_descriptions
+        latest.collect {|reg| reg.career_preferred_level.description }
+      end
+
+      def latest
+        all.select { |reg| reg.term_id == latest_term_id }
+      end
+
+      def latest_term_id
+        all.collect { |reg| reg.term_id }.sort.last
```

Comment:
I think we can simplify by sorting by term_id by default (also, let's memoize `#all`)

```
def all
  @all ||= data_feed.map do
    ::User::Academics::Registration.new(data)
  end.sort(&:term_id)
end

def latest
  all.last
end
```

Reply:
That's a great idea that will simplify the process of getting the latest term ID, however `#latest` is supposed to return multiple registrations if there are more than one for the latest term id (some students have two)